core/src/main/java/org/apache/iceberg/encryption/AesGcmOutputStream.java

```diff
@@ -158,4 +163,17 @@ private void encryptAndWriteBlock() throws IOException {
     positionInPlainBlock = 0;
     currentBlockIndex++;
   }
+
+  public static long gcmEncryptionLength(long plainLength) {
```

Comment:
todo: make private

Reply:
Do we really have to do this math? Can't we just check the bytes written? Haven't looked at the underlying code here so I could be lost. Just seems weird that we have to recalculate this

Reply:
Do we have test coverage for 1 block, x blocks and x + fractional blocks?

Reply:
> Can't we just check the bytes written? 

Yes we can! Excellent idea! For me, it was "straightforward" to re-use this util function I had from before; but replacing it with a direct call to the target stream length is WAY simpler and more intuitive.